src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -520,15 +522,18 @@ private void handleCloudWatchAttemptStatus(CloudWatchAttempt cloudWatchAttempt)
         // Update the runtime configuration and store the last processed file information
         context.runOnPublishQueueAndWait(() -> {
             processingFilesInformation.forEach((componentName, processingFiles) -> {
-                // Update old config value to handle downgrade from v 2.3.1 to older ones
-                Topics componentTopicsDeprecated =
-                        getRuntimeConfig().lookupTopics(PERSISTED_COMPONENT_CURRENT_PROCESSING_FILE_INFORMATION,
-                                componentName);
-
-                if (processingFiles.getMostRecentlyUsed() != null) {
-                    componentTopicsDeprecated.updateFromMap(
-                            processingFiles.getMostRecentlyUsed().convertToMapOfObjects(),
-                        new UpdateBehaviorTree(UpdateBehaviorTree.UpdateBehavior.REPLACE, System.currentTimeMillis()));
+                if (isDeprecatedVersionSupported()) {
```

Comment:
Didn't we check this in the parent if statement?

Reply:
No, that's a completely different method.

Reply:
it's confusing that they have the same signature then `isDeprecatedVersionSupported()`